eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Ratify.hs

```diff
@@ -203,11 +205,15 @@ ratifyTransition = do
 
   case rsig of
     ast :<| sigs -> do
-      let GovActionState {gasAction, gasExpiresAfter} = ast
-      if spoAccepted env ast && dRepAccepted env ast dRepThreshold
+      let GovActionState {gasId, gasAction, gasExpiresAfter} = ast
+      if spoAccepted env ast
+        && dRepAccepted env ast dRepThreshold
+        && prevActionAsExpected gasAction (ensPrevGovActionIds rsEnactState)
```

Comment:
This should be the first check since it is the fastest one to verify. This way we will short-circuit the if branch without redundantly looking through all the votes whenever prevActionId doesn't match.
```suggestion
      if prevActionAsExpected gasAction (ensPrevGovActionIds rsEnactState)
        && spoAccepted env ast
        && dRepAccepted env ast dRepThreshold
```